### ISSUE TITLE
provide a spec file for CentOS 7

### ISSUE THREAD

**Issue by mrbaseman** [adrienverge/openfortivpn#414]
I was asked to provide a spec file for a current _openfortivpn_ version. Currently we don't maintain distribution specific files in the upstream repository though. Anyhow, we have example files for ppp configuration in the etc subdirectory. Maybe we can put example spec files in a similar directory, or we could add a wiki page. Here is a proposal for the content of the spec file:
```
Name:           openfortivpn
Version:        1.8.0
Release:        1
Summary:        Client for PPP+SSL VPN tunnel services

Group:          Applications/Internet
License:        GPLv3+
URL:            https://github.com/adrienverge/openfortivpn
Source0:        openfortivpn-%{version}.tar.gz

BuildRequires:  autoconf
BuildRequires:  automake
BuildRequires:  openssl-devel
Requires:       ppp

%description
openfortivpn is a client for PPP+SSL VPN tunnel services. It spawns a pppd
process and operates the communication between the gateway and this process.

It is compatible with Fortinet VPNs.


%prep
%setup -q -n %{name}-%{version}


%build
autoreconf -fi
%configure
make %{?_smp_mflags} V=1


%install
make install DESTDIR=%{buildroot}


%files
%{_bindir}/openfortivpn
%{_mandir}/man1/openfortivpn.1*
%dir %{_sysconfdir}/openfortivpn
%config(noreplace) %{_sysconfdir}/openfortivpn/config
%{_datadir}/openfortivpn/config.template
%doc README.md
%license LICENSE
```
I'm sure there are other suggestions, as for example Fedora and OpenSUSE are rpm based distributions as well, and they provide packages for _openfortivpn_, too. 
Maybe we can provide a minimal template which mostly works on all rpm based platforms? 
However, I haven't looked at the details yet

**Comment by adrienverge** [comment]
I already packaged it for Fedora: https://bodhi.fedoraproject.org/updates/?packages=openfortivpn
The spec file can be copied-pasted from the SRPM :)

> Maybe we can provide a minimal template which mostly works on all rpm based platforms?

I'd rather let distribution files outside the repo, it's a lot of work to maintain. If we start with Fedora/CentOS, we'll have to do Debian-based distributions too, etc.

**Comment by mrbaseman** [comment]
That's a good point. I rather thought of a rough template as a starting point. But I must admit, one can find SRPMs as a starting point for different distributions, but the main work is adapting them for a specific distribution and maintaining all the details. The background for this was that there are no binary packages for CentOS, yet. @lkundrak has packaged more recent versions for Fedora. 
If I'm not mistaken the way to contribute to the CentOS EPEL repository is that a package maintainer who has provided to Fedora for some time [requests additional branches](https://fedoraproject.org/wiki/EPEL_Package_Maintainers) on EPEL. However, I'm not sure if this applies for the CentOS branches in EPEL too, or if this is a complete separate story. Maybe you guys who have already contributed to Fedora know better about those regulations.

**Comment by adrienverge** [comment]
openfortivpn is on its way to EPEL 7:
- package built and downloadable from Koji: https://koji.fedoraproject.org/koji/taskinfo?taskID=31412171
- EPEL update pending: https://bodhi.fedoraproject.org/updates/openfortivpn-1.8.0-1.el7

**Comment by mrbaseman** [comment]
great, and we have new architectures as well (at least I haven't tested yet on arm64 or power 7)

**Comment by mrbaseman** [comment]
@adrienverge, #415 is probably one of the packages that you have built with 1.8.0. 
There are two workarounds:
- install `pppd` on the build system, or
- specify `--with-pppd=/usr/sbin/pppd` in the spec file
We have a fix for this on the current master (#399)

**Comment by mrbaseman** [comment]
all the rpms are available under [this URL](https://koji.fedoraproject.org/koji/packageinfo?packageID=20983), currently in testing

**Comment by mrbaseman** [comment]
since the binary packages for epel are available I think we can close this now
